web/app/js/components/Octopus.jsx

```diff
@@ -115,22 +136,24 @@ class Octopus extends React.Component {
     }
 
     return (
-      <Grid item key={resource.type + "-" + resource.name} >
-        <Card className={type === "neighbor" ? classes.neighborNode : classes.centerNode} title={display}>
-          <CardContent>
-
-            <Typography variant={type === "neighbor" ? "subtitle1" : "h6"} align="center">
-              { this.linkedResourceTitle(resource, display) }
-            </Typography>
-
-            <Progress variant="determinate" value={resource.successRate * 100} />
-
-            <Table padding="dense">
-              {showTcp ? this.renderTCPStats(resource) : this.renderHttpStats(resource)}
-            </Table>
-          </CardContent>
-        </Card>
-      </Grid>
+      <RootRef rootRef={el => type === "neighbor" ? this.refCallback(el, index, isOutbound) : undefined} key={resource.type + "-" + resource.name}>
```

Comment:
There is a lot of logic in this line - is it possible to calculate `rootRef` before we get to the `return` statement, for easier debugging?